### ISSUE TITLE
Fix unstable Windows tests for the tests within 

### ISSUE THREAD

**Issue by Skeen** [fstab/grok_exporter#105]
...

    If you have any idea how os.Rename(...) can be successful but the file is still there after that, I'd be grateful, but this has nothing to do with your changes.

Are we sure that os.rename(...) is indeed atomic on Windows?

I don't believe that is the case in general, as Python calls MoveFileW() on Windows, which is atomic in specific cases, but not as a general case (such as with network mounted file-systems).
On Windows to ensure atomic renaming, you'd have to create a transaction with CreateTransaction() followed by a call to MoveFileTransacted(), and finally a commit.

So in general, I would not assume a os.rename call to be atomic on Windows.

Potential solutions:

    Find a library which does transactional file-moves, and accept that the tests will only pass on Windows Vista and higher.
    Add a sleep / busy wait loop, continuously checking whether the file has actually been moved yet, and leaving the loop after that happens.

**Comment by fstab** [comment]
I agree, we should fix it and not ignore it. Here's an example error message (there are other flaky things on Windows, but this one happens often):
```
--- FAIL: TestAll (76.82s)
    --- FAIL: TestAll/multiple_log_files,_single_directory(closeFileAfterEachLine,fseventTailer,create,cp) (0.01s)
        fswatcher_test.go:652: C:\Users\appveyor\AppData\Local\Temp\1\grok_exporter617375194\logdir still contains file logfile2.log after mv.
```
The error comes from [fswatcher_test.go#L482-L484](https://github.com/fstab/grok_exporter/blob/44b195e2dd91541c1625204c092ff4139b50747b/tailer/fswatcher_test.go#L482-L484).

The test uses different configurations to simulate logrotate configurations. The configuration is shown in `(...)` next to the test name. In this example the `cp` option is used, which means that the logfile is not really moved, but rather first copied and then the original is deleted. As the file is still there after that, it means that `os.Remove()` returned successfully, but didn't really remove the file. On Windows, `os.Remove()` is implemented using the `DeleteFileW` function.

Other configurations are flaky as well:

```
--- FAIL: TestAll (76.99s)
    --- FAIL: TestAll/multiple_log_files,_single_directory(closeFileAfterEachLine,fseventTailer,create,rm) (0.01s)
        fswatcher_test.go:652: C:\Users\appveyor\AppData\Local\Temp\1\grok_exporter609323269\logdir still contains file logfile2.log after mv.
```

Here the option is `mv`, meaning that an actual `os.Rename()` is called, which maps on Windows to the `MoveFileExW` function.

I'd find it weird if Windows file system operations were asynchronous, but we could try to implement a retry-with-timeout on [fswatcher_test.go#L482-L484](https://github.com/fstab/grok_exporter/blob/44b195e2dd91541c1625204c092ff4139b50747b/tailer/fswatcher_test.go#L482-L484) to see if that helps.
